packages/core/strapi/lib/commands/actions/develop/action.js

```diff
@@ -106,12 +106,28 @@ const primaryProcess = async ({ distDir, appDir, build, isTSProject, watchAdmin,
   cluster.fork();
 };
 
-const workerProcess = ({ appDir, distDir, watchAdmin, polling, isTSProject }) => {
-  const strapiInstance = strapi({
+const workerProcess = async ({ appDir, distDir, watchAdmin, polling, isTSProject }) => {
+  const strapiInstance = await strapi({
     distDir,
     autoReload: true,
     serveAdminPanel: !watchAdmin,
-  });
+  }).load();
+
+  const shouldGenerateTypeScriptTypes = strapiInstance.config.get(
+    'server.typescript.dev.generate',
+    true
```

Comment:
If we set the default to `false` we can target `main` and make this available as an experimental feature in the next release, and then flip the default to `true` in the big TS release once it's ready and documented